src/resolvers/tokenlists.ts

```diff
@@ -0,0 +1,16 @@
+import { resize } from '../utils';
+import { max } from '../constants.json';
+import { fetchHttpImage } from './utils';
+import { findImageUrl, updateExpiredAggregatedTokenList } from '../helpers/tokenlists';
+
+export default async function resolve(address: string, chainId: string) {
+  try {
+    await updateExpiredAggregatedTokenList();
```

Comment:
If we remove the `await` here:

- the first request triggering the token list refresh will always return immediately with nothing, with the list refreshing async
- a request waiting for list refresh will return immediately with current list, instead of waiting for new list, which is refreshing async